### PR TITLE
QoL improvements to add channel to team flow

### PR DESCRIPTION
**Pull request by KevLehman**
## Proposed changes (including videos or screenshots)

- Fixed canAccessRoom validation
- Added e2e tests
- Removed channels that user cannot add to the team from autocomplete suggestions
- Improved error messages

### CODE REVIEW

File: app/models/server/raw/Rooms.js [hunk 1]
```diff
@@ -204,8 +204,8 @@ export class RoomsRaw extends BaseRaw {
 				},
 			],
 			name: nameRegex,
+			'u._id': uid,
```
Comment (sampaiodiego) [inline app/models/server/raw/Rooms.js:207]: I know we used this filter on the other validation as well, but I don't think this is the correct way to validate what we need. this field actually stores the "creator" of the channel, which is formerly know as the owner, but we also have a owner role, which I think is more appropriate. Like when a user transfer the ownership of a channel we don't update the `u` field, we just give the new user the `owner` role.

the role is stored on Subscriptions, so here you could rely that filtered list of _ids where the user has the `owner` role..

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: app/models/server/raw/Rooms.js (hunk 1 above)
Comment (KevLehman) [reply app/models/server/raw/Rooms.js:207]: Yeah, I know about the `owner` role of the subs, I was trying to keep both filters the same (the one here and the one in the `addRooms` service.

If checking the subs is the way to go, I'll change this and the other :)

File: app/api/server/v1/teams.ts [hunk 2]
```diff
@@ -78,7 +78,7 @@ API.v1.addRoute('teams.addRooms', { authRequired: true }, {
 		}
 
 		if (!hasPermission(this.userId, 'add-team-channel', team.roomId)) {
-			return API.v1.unauthorized();
+			return API.v1.unauthorized('Teams_add_channel_no_permission');
```
[the remark this answers is not in the document]
Comment (KevLehman) [reply app/api/server/v1/teams.ts:81]: Ah, didn't know the `-` were meant to be error codes, I'll change it.###

File: app/models/server/raw/Rooms.js (hunk 1 above)
Comment (sampaiodiego) [reply app/models/server/raw/Rooms.js:207]: yep, I think that is the expected behavior.. I didn't realize that before 🙈

Comment (KevLehman) [reply app/models/server/raw/Rooms.js:207]: Yup, you're right on that 👀 we should've modified that validation before 😬

Comment (KevLehman) [reply app/models/server/raw/Rooms.js:207]: @sampaiodiego what should be the behavior of the autocomplete then? Currently, it fetches all public channels and private channels the user is part of. Should this be changed to only show channels where the user is the owner (disregard if it's public/private)?

Comment (sampaiodiego) [reply app/models/server/raw/Rooms.js]: Yep. I think so :)

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**